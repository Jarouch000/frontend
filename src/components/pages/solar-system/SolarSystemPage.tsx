import React from 'react'

import Links from '../../../links'
import Planets from './Planets'
import * as S from './styles'

/**
 * Represents content with an interactive demo of planetary and satellites datasets.
 */
const SolarSystemPage = () => (
  <>
    <S.SolarSystemHeader>
      This is an interactive version of <a href='https://nssdc.gsfc.nasa.gov/planetary/factsheet/'>Planetary Fact
      Sheet</a> and <a href='https://ssd.jpl.nasa.gov/?sat_phys_par'>Planetary Satellite Physical Parameters</a>
      &nbsp;datasets from {Links.jpl}.
      <p>
        Tips: <span className='highlight'>Select the planet below</span> to see only its satellites
        or <span className='highlight'>click on the column</span> to sort.
      </p>
    </S.SolarSystemHeader>

    <hr />

    <Planets />
  </>
)

export default SolarSystemPage
