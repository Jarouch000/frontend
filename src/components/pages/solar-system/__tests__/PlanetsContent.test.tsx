import React from 'react'
import { shallow } from 'enzyme'
import SolarSystemPage from '../SolarSystemPage'

describe('SolarSystemPage', () => {
  it('should render correctly', () => {
    expect(shallow(<SolarSystemPage />)).toMatchSnapshot()
  })
})
