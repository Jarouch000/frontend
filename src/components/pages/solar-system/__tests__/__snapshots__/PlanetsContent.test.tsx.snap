// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SolarSystemPage should render correctly 1`] = `
<Fragment>
  <styled.div>
    This is an interactive version of
     
    <a
      href="https://nssdc.gsfc.nasa.gov/planetary/factsheet/"
    >
      Planetary Fact Sheet
    </a>
     and
     
    <a
      href="https://ssd.jpl.nasa.gov/?sat_phys_par"
    >
      Planetary Satellite Physical Parameters
    </a>
     datasets from
     
    <a
      href="https://www.jpl.nasa.gov/"
    >
      NASA Jet Propulsion Laboratory
    </a>
    .
    <p>
      Tips: 
      <styled.span
        highlight={true}
      >
        Select the planet below
      </styled.span>
       to see only its satellites or
       
      <styled.span
        highlight={true}
      >
        click on the column
      </styled.span>
       to sort.
    </p>
  </styled.div>
  <hr />
  <Connect(Planets) />
</Fragment>
`;
